vendor/k8s.io/kubernetes/pkg/kubectl/cmd/set/set_image.go

```diff
@@ -134,13 +134,11 @@ func (o *ImageOptions) Complete(f *cmdutil.Factory, cmd *cobra.Command, args []s
 		ContinueOnError().
 		NamespaceParam(cmdNamespace).DefaultNamespace().
 		FilenameParam(enforceNamespace, o.Recursive, o.Filenames...).
+		SelectorParam(o.Selector).
+		ResourceTypeOrNameArgs(o.All, o.Resources...).
+		Latest().
```

Comment:
Latest() does a server fetch, right? I thought the point of Local was to avoid that